### PR TITLE
Feature/diff

### PR DESCRIPTION
**Pull request by tneuer**
### Summary

Implements the TimeSeries.diff() method wrapping around pandas.DataFrame.diff(). Often used for making TimeSeries stationary.

### Other Information

Additionally tests were added, the possibility to difference multiple times with the "n" parameter and to retain the "na" from pandas.DataFrame.diff() it its functionality wants to be reproduced.

### CODE REVIEW

File: darts/timeseries.py [hunk 1]
```diff
@@ -669,6 +669,39 @@ def shift(self, n: int) -> 'TimeSeries':
         new_series.index = new_time_index
         return TimeSeries(new_series, self.freq_str())
 
+    def diff(self, n: Optional[int] = 1,
```
Comment (hrzn) [inline darts/timeseries.py:672]: You can put each argument on a different line (including `self`).

File: darts/timeseries.py [hunk 2]
```diff
@@ -669,6 +669,39 @@ def shift(self, n: int) -> 'TimeSeries':
...
+        -------
+        TimeSeries
+            A TimeSeries constructed after differencing.
+        """
+        if not isinstance(n, int) or n < 1:
+             raise_log(ValueError("'n' must be a positive integer >= 1."))
+        if not isinstance(periods, int):
+             raise_log(ValueError("'periods' must be an integer."))
+
+        temp_df = self.pd_dataframe(copy=True)
```
Comment (hrzn) [inline darts/timeseries.py:698]: Not sure you need the copying if `diff()` already returns a copy and doesn't work in-place (to be confirmed but I guess that's the case).

File: darts/timeseries.py [hunk 3]
```diff
@@ -669,6 +669,39 @@ def shift(self, n: int) -> 'TimeSeries':
         new_series.index = new_time_index
         return TimeSeries(new_series, self.freq_str())
 
+    def diff(self, n: Optional[int] = 1,
+                periods: Optional[int] = 1,
+                dropna: Optional[bool] = True) -> 'TimeSeries':
+        """
+        Returns a differenced time series with the goal to make a time series stationary.
```
Comment (hrzn) [inline darts/timeseries.py:676]: I would slightly rephrase: `Returns a differenced time series. This is often used to make a time series stationary.`

**Review by hrzn — APPROVED** [review]
Few small comments but looks good otherwise, thanks!

File: darts/timeseries.py (hunk 2 above)
Comment (tneuer) [reply darts/timeseries.py]: Done

File: darts/timeseries.py (hunk 1 above)
Comment (tneuer) [reply darts/timeseries.py]: Done

File: darts/timeseries.py (hunk 3 above)
Comment (tneuer) [reply darts/timeseries.py]: And done

**Pull request merged**